File: Data Structures & Algorithms/reorder-linked-list/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public void reorderList(ListNode head) {
        if (head.next == null) {
            return;
        }
        ListNode tort = head;
        ListNode hare = head;
        while (hare != null && hare.next != null) {
            tort = tort.next;
            hare = hare.next.next;
        }
        ListNode current = tort.next;
        tort.next = null;
        ListNode prev = tort;
        System.out.println(current.val);

        
        ListNode temp = current;
        while (current != null) {
            current = current.next;
            temp.next = prev;
            prev = temp;
            temp = current;
        }
        
        System.out.println(prev.val);
        ListNode tail = prev;
        temp = head;
        System.out.println(temp.val);
        int choice = 1;
        while (tail != head) {
            if (choice == 1) {
                temp = head.next;
                head.next = tail;
                head = temp;
                choice = 0;
            }
            else {
                temp = tail.next;
                tail.next = head;
                tail = temp;
                choice = 1;
            }

        }
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        current = prices[0] 
        profit = 0
        i = 0
        sell = 0
        while (i < len(prices)):
            if (prices[i] >= current):
                if (sell < prices[i]):
                    sell = prices[i]
            else:
                temp = sell - current
                current = prices[i]
                sell = 0
                if (temp > profit):
                    profit = temp
            i += 1
        temp = sell - current
        if (temp > profit):
            profit = temp
        return profit
        